adapters/criteo/criteo.go

```diff
@@ -93,6 +109,21 @@ func (a *adapter) MakeBids(request *openrtb2.BidRequest, requestData *adapters.R
 		}
 	}
 
+	if response.Ext != nil {
+		var responseExt CriteoExt
+		if err := json.Unmarshal(response.Ext, &responseExt); err == nil && responseExt.Igi != nil {
+			bidResponse.FledgeAuctionConfigs = make([]*openrtb_ext.FledgeAuctionConfig, 0, len(responseExt.Igi))
+			for _, igi := range responseExt.Igi {
+				fledgeAuctionConfig := &openrtb_ext.FledgeAuctionConfig{
+					ImpId:  igi.ImpId,
+					Bidder: a.bidderName,
+					Config: igi.Igs[0].Config,
+				}
+				bidResponse.FledgeAuctionConfigs = append(bidResponse.FledgeAuctionConfigs, fledgeAuctionConfig)
+			}
+		}
+	}
```

Comment:
Can we move this entire block of code into its own function and add a tests to cover edge cases such as `responseExt.Igi`, `igi.Igs`, `igi.ImpId`, or `igi.Igs[0].Config` being empty?
```
111
112 -     if response.Ext != nil {
113 -         var responseExt CriteoExt
114 -         if err := json.Unmarshal(response.Ext, &responseExt); err == nil && responseExt.Igi != nil {
115 -             bidResponse.FledgeAuctionConfigs = make([]*openrtb_ext.FledgeAuctionConfig, 0, len(responseExt.Igi))
116 -             for _, igi := range responseExt.Igi {
117 -                 fledgeAuctionConfig := &openrtb_ext.FledgeAuctionConfig{
118 -                     ImpId:  igi.ImpId,
119 -                     Bidder: a.bidderName,
120 -                     Config: igi.Igs[0].Config,
121 -                 }
122 -                 bidResponse.FledgeAuctionConfigs = append(bidResponse.FledgeAuctionConfigs, fledgeAuctionConfig)
123 -             }
124 -         }
125 -     }
    +     bidResponse.FledgeAuctionConfigs = append(bidResponse.FledgeAuctionConfigs, buildFledgeAuctionConfigsList(response.Ext))
126  
127       return bidResponse, nil
128   }
adapters/criteo/criteo.go
```

Reply:
I added test cases to cover all these edge cases, thanks.